src/Iam.php

```diff
@@ -102,9 +102,52 @@ public function signBlob($email, $accessToken, $stringToSign, array $delegates =
             Utils::streamFor(json_encode($body))
         );
 
-        $res = $httpHandler($request);
+        $res = ($this->httpHandler)($request);
         $body = json_decode((string) $res->getBody(), true);
 
         return $body['signedBlob'];
     }
+
+    /**
+     * Sign a string using the IAM signBlob API.
+     *
+     * Note that signing using IAM requires your service account to have the
+     * `iam.serviceAccounts.signBlob` permission, part of the "Service Account
+     * Token Creator" IAM role.
+     *
+     * @param string $clientEmail The service account email.
+     * @param string $targetAudience The audience for the ID token.
+     * @param string $bearerToken The token to authenticate the IAM request.
+     *
+     * @return string The signed string, base64-encoded.
+     */
+    public function generateIdToken(
+        string $clientEmail,
+        string $targetAudience,
+        string $bearerToken
```

Comment:
perhaps there could be a fourth parameter `array $headers = []`. This way the calling class (e.g. `ImpersonatedServiceAccountCredentials`) could add a token endpoint metrics header. (The bearer token could stay as-is as it is required for the id token to be returned.)

Reply:
Thanks for this feedback! Could you provide me an example of what that would look like? I assume it would look like adding the following in the call to [Iam::generateIdToken](https://github.com/googleapis/google-auth-library-php/pull/581/files#diff-248500c0207a8c4941997a322cc9895767971446b14eb6722257d5c694c3c7ffR239):

```php
            $idToken = (new Iam($httpHandler, $this->getUniverseDomain()))->generateIdToken(
                $this->auth->getIssuer(),
                $this->auth->getAdditionalClaims()['target_audience'],
                $jwt,
                $this->applyTokenEndpointMetrics([], 'it') // this is the new line
            );
```

Reply:
Yes, that is exactly what I had in mind. All requests I encountered getting an acces or id token send metrics headers, so it makes sense to add them here too. The implementation would be something like this.
```
    /**
     * @param array $headers Optional headers to send to the api, defaults to an empty array
     */
    public function generateIdToken(
        string $clientEmail,
        string $targetAudience,
        string $bearerToken,
        array $headers = [] // -> added optional parameter
): string {
        // ....
       // updated header assignment
       $headers['Authorization'] = 'Bearer ' . $bearerToken; 
        // ....
```

Reply:
This has been added in  475bc74